lib/lambda_ethereum_consensus/validator/validator.ex

```diff
@@ -230,7 +231,7 @@ defmodule LambdaEthereumConsensus.Validator do
         old -> old |> Enum.reverse() |> Enum.take(1)
       end
 
-    %{duties | attester: attester_duties, proposer: old_duty ++ proposer_duties}
+    %{duties | attester: attester_duties, proposer: Enum.dedup(old_duty ++ proposer_duties)}
```

Comment:
This reintroduces a bug related to last-slot-in-epoch proposals. We should instead discard "future" duties before computing `old_duty`. Something like:

```elixir
slot = Misc.compute_start_slot_for_epoch(epoch)
case duties.proposer do
  :not_computed -> []
  old -> old |> Enum.filter(&(&1 < slot)) |> Enum.reverse() |> Enum.take(1)
end
```